src/components/structures/GroupView.js

```diff
@@ -18,10 +18,115 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import MatrixClientPeg from '../../MatrixClientPeg';
 import sdk from '../../index';
+import dis from '../../dispatcher';
 import { sanitizedHtmlNode } from '../../HtmlUtils';
 import { _t } from '../../languageHandler';
+import AccessibleButton from '../views/elements/AccessibleButton';
 
 
+function categoryRoomListNode(rooms, categoryId, category) {
+    const roomNodes = rooms.map((r) => {
+        return <FeaturedRoom key={r.room_id} summaryInfo={r} />;
+    });
+    let catHeader = null;
+    if (category && category.profile) {
+        catHeader = <div className="mx_GroupView_featuredThings_category">{category.profile.name}</div>;
+    }
+    return <div key={categoryId}>
+        {catHeader}
+        {roomNodes}
+    </div>;
+}
+
+const FeaturedRoom = React.createClass({
+    displayName: 'FeaturedRoom',
+
+    props: {
+        summaryInfo: PropTypes.object.isRequired,
+    },
+
+    onClick: function(e) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        dis.dispatch({
+            action: 'view_room',
+            room_alias: this.props.summaryInfo.profile.canonical_alias,
+            room_id: this.props.summaryInfo.room_id,
+        });
+    },
+
+    render: function() {
+        const RoomAvatar = sdk.getComponent("avatars.RoomAvatar");
+
+        const oobData = {
+            roomId: this.props.summaryInfo.room_id,
+            avatarUrl: this.props.summaryInfo.profile.avatar_url,
+            name: this.props.summaryInfo.profile.name,
+        };
+        let permalink = null;
+        if (this.props.summaryInfo.profile && this.props.summaryInfo.profile.canonical_alias) {
+            permalink = 'https://matrix.to/#/' + this.props.summaryInfo.profile.canonical_alias;
+        }
+        let roomNameNode = null;
+        if (permalink) {
+            roomNameNode = <a href={permalink} onClick={this.onClick} >{this.props.summaryInfo.profile.name}</a>;
+        } else {
+            roomNameNode = <span>{this.props.summaryInfo.profile.name}</span>;
+        }
+
+        return <AccessibleButton className="mx_GroupView_featuredThing" onClick={this.onClick}>
+            <RoomAvatar oobData={oobData} width={64} height={64} />
+            <div className="mx_GroupView_featuredThing_name">{roomNameNode}</div>
+        </AccessibleButton>;
+    },
+});
+
+function roleUserListNode(users, roleId, role) {
+    const userNodes = users.map((u) => {
+        return <FeaturedUser key={u.user_id} summaryInfo={u} />;
+    });
+    let roleHeader = null;
+    if (role && role.profile) {
+        roleHeader = <div className="mx_GroupView_featuredThings_category">{role.profile.name}</div>;
+    }
+    return <div key={roleId}>
+        {roleHeader}
+        {userNodes}
+    </div>;
+}
+
+const FeaturedUser = React.createClass({
+    displayName: 'FeaturedUser',
+
+    props: {
+        summaryInfo: PropTypes.object.isRequired,
```

Comment:
It looks like you only ever use the `user_id`, although I'm guessing you'll want more stuff later?